contracts/ProcedureTable.sol

```diff
@@ -87,46 +80,63 @@ library ProcedureTable {
         // The second storage location (1) is used to store the address of the
         // contract.
         p.location = address(_get(0, pPointer + 1));
-        // TODO: add the capability list here. For now we only have a single
-        // capability slot for a single capability type, which is write.
         // For now this is:
         // type + key + value
         //  8   + 32  +  32   = 72 bytes
-        // Actually, for now it is simply a boolean to determine if the procedure
-        // is allowed to write or not
-
-        // The type of capability (currently only WRITE 0x7)
-        p.capabilityType = uint8(_get(0, pPointer + 2));
-        // The key to which we can write
-        p.capabilityKey = uint256(_get(0, pPointer + 3));
-        // The number of additional keys we can write to
-        p.capabilitySize = uint256(_get(0, pPointer + 4));
+        // Actually, for now it is simply a boolean to determine if the
+        // procedure is allowed to write or not
+        // TODO: I don't like the way we remove information here. If we want to
```

Comment:
Could you elaborate on this?

Reply:
This is fixed now, we currently serialise and deserialise all caps.